[Docs] Batch09: only `lookup` (#16878)
apache/druid: docs/querying/sql-functions.md

## LOOKUP

> Searches for `expr` in a registered [query-time lookup table](lookups.md) named `lookupName`. Returns `replaceMissingValueWith`, which defaults to `null` if `expr` is null or if the lookup doesn't contain a key for `expr`.

Reviewer: Let's move the conditional to the start of the sentence so it doesn't look like the expected return behavior.
```suggestion
Searches for `expr` in a registered [query-time lookup table](lookups.md) named `lookupName` and returns the mapped value. If `expr` is null or not contained in the lookup, returns `replaceMissingValueWith` if supplied, otherwise null.
```

Reviewer: Might need to update the table description too? Double check this

Author: Table description is just introducing the table that is being used, not how it is being used, I do not think it is necessary to update it

Author: Did need to be updated, changes are now live

> Look up `expr` in a registered [query-time lookup table](lookups.md) named `lookupName`. Returns `replaceMissingValueWith`, which defaults to `null`, when the `expr` is null or when the lookup does not contain a value for `expr`.

Reviewer 2: Other examples use third-person singular form of the verb in simple present tense (a mouthful). "Looks up" doesn't sound great. Maybe "searches for" is a better choice? 
```suggestion
Searches for `expr` in a registered [query-time lookup table](lookups.md) named `lookupName`. Returns `replaceMissingValueWith`, which defaults to `null` if `expr` is null or if the lookup doesn't contain a value for `expr`.
```

> Look up `expr` in a registered [query-time lookup table](lookups.md) named `lookupName`. Returns `replaceMissingValueWith`, which defaults to `null`, when the `expr` is null or when the lookup does not contain a value for `expr`.

Reviewer: >does not contain a value

Change "value" to "key" for more specificity

Author: switched "value" -> "key"

Author: Also switched "value" -> "key" on line 1096 for consistency

* **Syntax:** `LOOKUP(expr, lookupName[, replaceMissingValueWith])`
* **Function type:** Scalar, string

<details><summary>Example</summary>

> The following example uses a map lookup table, named `acronym_to_name` made with the following key-value pairs:

Reviewer 2: ```suggestion
The following example uses a map lookup table named `acronym_to_name`, made with the following key-value pairs:
```

Reviewer: >map lookup table

`map`-typed lookup

> The following example uses a `map`-typed lookup table named `acronym_to_name`, made with the following key-value pairs:

Reviewer: ```suggestion
The following example uses a `map`-typed lookup table named `acronym_to_name`, which contains the following key-value pairs:
```

```json
{
	“SJU”: Luis Munoz Marin International Airport”,

> 	“IAD”: “Dulles International AirportDulles International Airport”

Reviewer: Can you update this to
* Change tabs to spaces
* Used non-stylized quotation marks
* Add missing quotation marks for the first line

Author: - [x] Change tabs to spaces
- [x] Used non-stylized quotation marks
- [x] Add missing quotation marks for the first line

}
```

> The example uses `acronym_to_name` to map the `Origin` column from the `flight-carriers` datasource to its full airport name. Returns `NaN` if no matching value exist in the lookup table.

Reviewer 2: ```suggestion
The example uses `acronym_to_name` to map the `Origin` column from the `flight-carriers` datasource to the corresponding full airport name. Returns `NaN` if no matching value exists in the lookup table.
```

> The example uses `acronym_to_name` to map the `Origin` column from the `flight-carriers` datasource to its full airport name. Returns `NaN` if no matching value exist in the lookup table.

Reviewer: Is `NaN` a good default value to use, when each lookup key-value pair is already not a number?

Author: In retrospect probably non, I could change it to "NULL"

## LOOKUP

Searches for `expr` in a registered [query-time lookup table](lookups.md) named `lookupName` and returns the mapped value. If `expr` is null or not contained in the lookup, returns `replaceMissingValueWith` if supplied, otherwise returns null.

> * **Syntax:** `LOOKUP(expr, lookupName[, replaceMissingValueWith])`

Reviewer 3: Can we use the name `defaultValue` for the 3rd parameter? I think it would be easier to understand.

* **Function type:** Scalar, string

<details><summary>Example</summary>

> The following example uses a `map`-typed lookup table named `acronym_to_name`, which contains the following key-value pairs:

Reviewer 4: ```suggestion
The following example uses a `map` type lookup table named `acronym_to_name`, which contains the following key-value pairs:
```
hyphenated construction unnecessary
